lib/tasks/docker.rake

```diff
@@ -14,6 +14,8 @@
 # Other useful environment variables (not specific to this rake task)
 # => COMMIT_HASH    used by the discourse_test docker image to load a specific commit of discourse
 #                   this can also be set to a branch, e.g. "origin/tests-passed"
+# => SKIP_LINT      if using the discourse_test docker image, this will skip the docker:test task, and only run the docker:lint task
```

Comment:
Using environment variables will make it much easier to change when running inside a docker container in CI, so how about `script/docker_test.rb` simply runs `rake docker:test`

Then inside the `rake docker:test` task we have both linting and testing. The linting bit can be surrounded by `unless ENV['SKIP_LINT']`, and the testing bit can be surrounded by `unless ENV['SKIP_TESTS']`